x-pack/plugins/grokdebugger/public/render_app.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+
+import { GrokDebugger } from './components/grok_debugger';
+import { GrokdebuggerService } from './services/grokdebugger/grokdebugger_service';
+import { I18nProvider } from '@kbn/i18n/react';
+import { KibanaContextProvider } from '../../../../src/plugins/kibana_react/public';
+import { InactiveLicenseSlate } from './components/inactive_license';
+
+export function renderApp(license, element, coreStart) {
+  if (!license.isActive) {
+    render(
+      <I18nProvider>
+        <InactiveLicenseSlate license={license} />
+      </I18nProvider>,
+      element
+    );
+    return () => unmountComponentAtNode(element);
+  } else {
+    render(
+      <KibanaContextProvider services={{ ...coreStart }}>
+        <I18nProvider>
+          <GrokDebugger grokdebuggerService={new GrokdebuggerService(coreStart.http)} />
+        </I18nProvider>
+      </KibanaContextProvider>,
+      element
+    );
+    return () => unmountComponentAtNode(element);
+  }
```

Comment:
Small nit: I think this branch will be easier to parse if we extract out the duplicated render call and return value.

```jsx
const content = license.isActive ? (
  <KibanaContextProvider services={{ ...coreStart }}>
    <I18nProvider>
      <GrokDebugger grokdebuggerService={new GrokdebuggerService(coreStart.http)} />
    </I18nProvider>
  </KibanaContextProvider>
) : (
  <I18nProvider>
    <InactiveLicenseSlate license={license} />
  </I18nProvider>
);

render(content, element);

return () => unmountComponentAtNode(element);
```

Reply:
Fair 👍 I'll change this.